File: growfit/ui/screens/new_exercise_screen.py
# new_exercise_screen.py

from kivy.uix.screenmanager import Screen
import growfit.core.utils as uc

class NewExerciseScreen(Screen):
    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        self.reps_counter = 12  # Initial value for repetitions
        self.sets_counter = 0  # Initial value for sets
        self.build_ui()

    def build_ui(self):
        from kivy.uix.boxlayout import BoxLayout
        from kivy.uix.label import Label
        from kivy.uix.button import Button
        layout = BoxLayout(orientation='vertical', padding=20, spacing=10)
        self.add_exercise_elements(layout)
        self.add_widget(layout)

    def add_exercise_elements(self, layout):
        from kivy.uix.label import Label
        from kivy.uix.button import Button
        self.sets_label = Label(text=f"Sets: {self.sets_counter}", font_size=30)
        layout.add_widget(self.sets_label)
        self.reps_label = Label(text=f"Repetitions: {self.reps_counter}", font_size=30)
        layout.add_widget(self.reps_label)
        layout.add_widget(Button(text="+1 Rep", size_hint=(1, 0.3), on_press=self.increment_reps))
        layout.add_widget(Button(text="-1 Rep", size_hint=(1, 0.3), on_press=self.decrement_reps))
        layout.add_widget(Button(text="Reset Reps", size_hint=(1, 0.3), on_press=self.reset_reps))
        layout.add_widget(Button(text="+1 Set", size_hint=(1, 0.3), on_press=self.increment_sets))
        layout.add_widget(Button(text="-1 Set", size_hint=(1, 0.3), on_press=self.decrement_sets))
        layout.add_widget(Button(text="Back", size_hint=(1, 0.3), on_press=self.go_back))

    def increment_reps(self, instance):
        self.reps_counter = uc.increment_counter(self.reps_counter)
        self.reps_label.text = f"Repetitions: {self.reps_counter}"

    def decrement_reps(self, instance):
        self.reps_counter = uc.reduce_counter(self.reps_counter)
        self.reps_label.text = f"Repetitions: {self.reps_counter}"

    def reset_reps(self, instance):
        self.reps_counter = 12
        self.reps_label.text = f"Repetitions: {self.reps_counter}"

    def increment_sets(self, instance):
        self.sets_counter += 1
        self.sets_label.text = f"Sets: {self.sets_counter}"

    def decrement_sets(self, instance):
        if self.sets_counter > 0:
            self.sets_counter -= 1
        self.sets_label.text = f"Sets: {self.sets_counter}"

    def go_back(self, instance):
        self.manager.current = 'new_routine'

File: growfit_flutter/lib/screens/new_exercise_screen.dart

/// Screen for creating a new exercise.
/// Allows the user to set sets and repetitions, and navigate back.

import 'package:flutter/material.dart';
import 'package:growfit_flutter/logic/counter_utils.dart';

/// Pantalla para crear un nuevo ejercicio.
/// Permite modificar sets y repeticiones, y volver atrás.
class NewExerciseScreen extends StatefulWidget {
  const NewExerciseScreen({super.key});

  @override
  _NewExerciseScreenState createState() => _NewExerciseScreenState();
}

class _NewExerciseScreenState extends State<NewExerciseScreen> {

  final TextEditingController _exerciseNameController = TextEditingController();
  int repsCounter = 12;
  int setsCounter = 3;

  void incrementReps() {
    setState(() {
      repsCounter = incrementCounter(repsCounter);
    });
  }

  void decrementReps() {
    setState(() {
      repsCounter = reduceCounter(repsCounter);
    });
  }

  void resetReps() {
    setState(() {
      repsCounter = 12;
    });
  }

  void incrementSets() {
    setState(() {
      setsCounter = incrementCounter(setsCounter);
    });
  }

  void decrementSets() {
    setState(() {
      setsCounter = reduceCounter(setsCounter);
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('New Exercise')),
      body: Padding(
        padding: const EdgeInsets.all(20.0),
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [

            // Input field for exercise name
            TextField(
              controller: _exerciseNameController,
              textAlign: TextAlign.center,
              style: const TextStyle(fontSize: 28),
              maxLength: 28,
              decoration: const InputDecoration(
                hintText: 'Enter exercise name',
                border: OutlineInputBorder(),
                contentPadding: EdgeInsets.symmetric(vertical: 10),
                counterText: '',
              ),
            ),

            // Display current number of sets
            Text(
              'Sets: $setsCounter',
              textAlign: TextAlign.center,
              style: const TextStyle(fontSize: 30),
            ),
            const SizedBox(height: 10),

            // Display current number of repetitions
            Text(
              'Repetitions: $repsCounter',
              textAlign: TextAlign.center,
              style: const TextStyle(fontSize: 30),
            ),
            const SizedBox(height: 20),

            // Button to increment repetitions
            ElevatedButton(
              onPressed: incrementReps,
              style: ElevatedButton.styleFrom(padding: const EdgeInsets.symmetric(vertical: 18)),
              child: const Text('+1 Rep'),
            ),

            // Button to decrement repetitions
            ElevatedButton(
              onPressed: decrementReps,
              style: ElevatedButton.styleFrom(padding: const EdgeInsets.symmetric(vertical: 18)),
              child: const Text('-1 Rep'),
            ),

            // Button to reset repetitions to default value
            ElevatedButton(
              onPressed: resetReps,
              style: ElevatedButton.styleFrom(padding: const EdgeInsets.symmetric(vertical: 18)),
              child: const Text('Reset Reps'),
            ),

            // Button to increment sets
            ElevatedButton(
              onPressed: incrementSets,
              style: ElevatedButton.styleFrom(padding: const EdgeInsets.symmetric(vertical: 18)),
              child: const Text('+1 Set'),
            ),

            // Button to decrement sets
            ElevatedButton(
              onPressed: decrementSets,
              style: ElevatedButton.styleFrom(padding: const EdgeInsets.symmetric(vertical: 18)),
              child: const Text('-1 Set'),
            ),

            // Button to go back to the previous screen
            ElevatedButton(
              onPressed: () {
                Navigator.pushNamedAndRemoveUntil(context, '/new_routine', (route) => false);
              },
              style: ElevatedButton.styleFrom(
                padding: const EdgeInsets.symmetric(vertical: 18),
                backgroundColor: Colors.grey,
              ),
              child: const Text('Back'),
            ),
          ],
        ),
      ),
    );
  }
}
